### PR TITLE
Discover v2 on Desktop

### LINKED ISSUE #3188: [Feature]: Discover v2 on Desktop
### Library name

@ledgerhq/ledger-live-desktop

### Describe your feature request

Port Discover v2 from mobile to desktop.

### PR DESCRIPTION
**Pull request by JunichiSugiura**
### 📝 Description

Port Discover v2 features from mobile to desktop.

[design](https://www.figma.com/file/iiV6KAXSqThtGQNLdFJYuW/LLD-%2F-Discover?node-id=2127-31863&t=ijhKrg3PSUebF9HK-0)

### ❓ Context

- **Impacted projects**: `ledger-live-mobile`, `ledger-live-desktop`, `live-common` 
- **Linked resource(s)**: close #3188  

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

### Tasks
- [x] toggle version based on discover feature flag
- [x] Port custom hooks from mobile to common (*: requires platform specific logic)
  - [x] `useCategories`
  - [x] `useDisclaimer`: *`route`
  - [x] `useSearch`: *`inputRef`
  - [x] `useRecentlyUsed`: *`usePlatfromState` (storage)
- [x] Catalog v2 UI
  - [x] Search bar
  - [x] Recently used
  - [x] Browse (WIP: label e.g. "Soon")

### CODE REVIEW

File: apps/ledger-live-mobile/src/screens/Platform/index.tsx [hunk 1]
```diff
@@ -18,6 +18,7 @@ export function Catalog(props: CatalogProps) {
 export function LiveApp(props: LiveAppProps) {
   const config = useFeature("discover");
 
+  // TODO: both version looks the same now
   return config?.enabled && config?.params.version === "2" ? (
     <LiveApp2 {...props} />
   ) : (
```
Comment (JunichiSugiura) [inline apps/ledger-live-mobile/src/screens/Platform/index.tsx:24]: I don't see a diff between LiveApp1 and LiveApp2. TODO: Check if it's fine to ditch 🤔

File: apps/ledger-live-desktop/src/renderer/screens/platform/LiveApp.tsx [hunk 2]
```diff
@@ -27,10 +28,11 @@ type Props = {
     customDappUrl?: string;
   };
 };
-export default function PlatformApp({ match, appId: propsAppId, location }: Props) {
+
+export function LiveApp({ match, appId: propsAppId, location }: Props) {
   const history = useHistory();
   const { params: internalParams, search } = location;
-  const { state: urlParams, customDappUrl } = useLocation();
+  const { state: urlParams, customDappUrl } = useLocation<{ returnTo: string }>();
```
Comment (JunichiSugiura) [inline apps/ledger-live-desktop/src/renderer/screens/platform/LiveApp.tsx:35]: - Where does `customDappUrl` come from?
- Is it Electron's deeplink, react-router-dom or both?
- What's the difference between `Prop["location"]` and return value of `useLocation`?

**Review by gre — COMMENTED** [review]
👋  we just merged https://github.com/LedgerHQ/ledger-live/pull/3359 which greatly improved the TypeScript checks on Ledger Live Desktop projects (components, modals, screens) and effectively make it type check on the CI. Could you rebase with it and ensure there is no issue on `pnpm desktop typecheck`? Thanks and don't hesitate to ask for any TS help.

**Comment by JunichiSugiura** [comment]
Oops, need to past the playwright test

**Comment by adrienlacombe-ledger** [comment]
@JunichiSugiura how is LLM impacted?

**Comment by JunichiSugiura** [comment]
@adrienlacombe-ledger I needed to tweak common hooks which has already been used in mobile but not fully compatible with desktop.

**Comment by gre** [comment]
👋  we have recently merged https://github.com/LedgerHQ/ledger-live/pull/3455 in develop branch which reinforces the TypeScript checks, mostly related to modals and coin families. Could you rebase and check that everything still pass?

Thanks,
gre.

**Comment by gre** [comment]
looks like you already did 👏

**Review by ComradeAERGO — APPROVED** [review]
Looking good to me, the code seems clean

I'd recommend another validation from @IAmMorrow or @Justkant though, as I a not as familiar as them with the whole Live's Architecture and specifics

File: libs/ledger-live-common/src/wallet-api/react.ts [hunk 3]
```diff
@@ -760,3 +768,218 @@ export enum ExchangeType {
...
+  );
+
+  return {
+    manifests,
+    categories,
+    manifestsByCategories,
+    selected,
+    setSelected,
+    reset,
+  };
```
Comment (Justkant) [inline libs/ledger-live-common/src/wallet-api/react.ts:804]: We might want to take a look at the usage of the hooks and if we need to add a `useMemo` on the returned value

File: libs/ledger-live-common/src/wallet-api/types.ts [hunk 4]
```diff
@@ -61,3 +62,12 @@ export type ConvertToLiveTransaction<
...
 > = (tx: T) => Partial<U>;
+
+export interface DiscoverDB {
+  recentlyUsed: RecentlyUsed[];
+}
+
+export interface RecentlyUsed {
+  id: string;
+  usedAt: string;
+}
```
Comment (Justkant) [inline libs/ledger-live-common/src/wallet-api/types.ts:73]: You might want to use `type` instead of `interface` here to avoid unwanted merge by ts

**Review by Justkant — COMMENTED** [review]
Some small stuff to look at but should be good to go, still need to test mobile

**Review by Justkant — APPROVED** [review]

**Pull request merged**